### PR TITLE
Tests order shopping cart

### PR DESCRIPTION
**Pull request by Den-moroz**

### CODE REVIEW

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Don't know why but here are changes from all PRs in this PR. Resolve it please so we can review only the logic related to tests of ShoppingCart functionality
image

File: src/main/java/core/basesyntax/bookstore/service/impl/OrderItemServiceImpl.java [hunk 1]
```diff
@@ -27,6 +29,10 @@ public List<OrderItemDto> getAllByOrderId(Long orderId) {
     @Override
     @Transactional
```
Comment (vovasalyha) [inline src/main/java/core/basesyntax/bookstore/service/impl/OrderItemServiceImpl.java]:
```suggestion
    @Transactional(readOnly = true)
```

File: src/test/java/core/basesyntax/bookstore/cartitem/CartItemRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        }
+    }
+
+    @Test
+    @DisplayName("Find cart items by shopping cart id")
+    @Sql(scripts = "classpath:database/cartitem/setup-cart-item-repository.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    void findCartItemsByShoppingCartId_validShoppingCartId_returnOneItem() {
+        Set<CartItem> actual = cartItemRepository.findCartItemsByShoppingCartId(1L);
+        Assertions.assertNotNull(actual);
```
Comment (vovasalyha) [inline src/test/java/core/basesyntax/bookstore/cartitem/CartItemRepositoryTest.java]: does repo return null or empty collection?

File: src/test/java/core/basesyntax/bookstore/cartitem/CartItemRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        Assertions.assertEquals(Set.of(VALID_CART_ITEM), actual);
+    }
+}
```
Comment (vovasalyha) [inline src/test/java/core/basesyntax/bookstore/cartitem/CartItemRepositoryTest.java:100]: is it all test cases? Maybe add some tests when zero items found and verify correct value is returned

File: src/test/java/core/basesyntax/bookstore/cartitem/CartItemServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    void save_validRequest_returnDto() {
+        when(bookRepository.getById(Mockito.anyLong())).thenReturn(DEFAULT_BOOK);
+        when(userService.getUser()).thenReturn(DEFAULT_USER);
+        when(shoppingCartRepository.findById(Mockito.anyLong()))
+                .thenReturn(Optional.of(DEFAULT_SHOPPING_CART));
+        when(cartItemMapper.toDto(Mockito.any())).thenReturn(VALID_RESPONSE_DTO);
+        when(cartItemRepository.save(Mockito.any())).thenReturn(Mockito.any());
+
+        CartItemDto actual = cartItemService.save(VALID_REQUEST);
+        Assertions.assertNotNull(actual);
```
Comment (vovasalyha) [inline src/test/java/core/basesyntax/bookstore/cartitem/CartItemServiceTest.java]: add static import for Assertions

File: src/test/java/core/basesyntax/bookstore/cartitem/CartItemServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    @Test
+    @DisplayName("Verify save() method")
+    void save_validRequest_returnDto() {
+        when(bookRepository.getById(Mockito.anyLong())).thenReturn(DEFAULT_BOOK);
+        when(userService.getUser()).thenReturn(DEFAULT_USER);
+        when(shoppingCartRepository.findById(Mockito.anyLong()))
+                .thenReturn(Optional.of(DEFAULT_SHOPPING_CART));
+        when(cartItemMapper.toDto(Mockito.any())).thenReturn(VALID_RESPONSE_DTO);
```
Comment (vovasalyha) [inline src/test/java/core/basesyntax/bookstore/cartitem/CartItemServiceTest.java]: add static import for Mockito

File: src/test/java/core/basesyntax/bookstore/cartitem/CartItemServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        when(cartItemRepository.save(Mockito.any())).thenReturn(Mockito.any());
+
+        CartItemDto actual = cartItemService.save(VALID_REQUEST);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(VALID_RESPONSE_DTO, actual);
+    }
```
Comment (vovasalyha) [inline src/test/java/core/basesyntax/bookstore/cartitem/CartItemServiceTest.java:126]: verify all methods that are expected are called and add some test case for edge cases, not only success case

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Generally everything is fine but I see that you do not test cases other than success in your service unit tests, please add edge cases too.

File: src/test/java/core/basesyntax/bookstore/cartitem/CartItemRepositoryTest.java (hunk 2 above)
Comment (Den-moroz) [reply src/test/java/core/basesyntax/bookstore/cartitem/CartItemRepositoryTest.java]: empty collection and I check it using assertNotNull, what's wrong?

**Review by nacenik — APPROVED** [review]
LGTM! 👍
